app/src/main/java/org/simple/clinic/drugs/EditMedicinesUiRenderer.kt

```diff
@@ -10,14 +10,39 @@ class EditMedicinesUiRenderer(private val ui: EditMedicinesUi) : ViewRenderer<Ed
 
   override fun render(model: EditMedicinesModel) {
     if (model.prescribedDrugs != null && model.protocolDrugs != null)
-      renderPrescribedProtocolDrugs(model.prescribedDrugs, model.protocolDrugs)
+      renderPrescribedProtocolDrugs(model, model.prescribedDrugs, model.protocolDrugs)
   }
 
-  private fun renderPrescribedProtocolDrugs(prescribedDrugs: List<PrescribedDrug>, protocolDrugs: List<ProtocolDrugAndDosages>) {
-    val prescribedProtocolDrugs = prescribedDrugs.filter { it.isProtocolDrug }
-    val isAtLeastOneCustomDrugPrescribed = prescribedDrugs.any { it.isProtocolDrug.not() }
+  private fun renderPrescribedProtocolDrugs(
+      model: EditMedicinesModel,
+      prescribedDrugs: List<PrescribedDrug>,
+      protocolDrugs: List<ProtocolDrugAndDosages>
+  ) {
+    val (prescribedProtocolDrugs, prescribedCustomDrugs) = prescribedDrugs.partition(model::isProtocolDrug)
+    val isAtLeastOneCustomDrugPrescribed = prescribedDrugs.any { prescribedDrug -> model.isProtocolDrug(prescribedDrug).not() }
```

Comment:
This check can be simplified:

```kotlin
val isAtLeastOneCustomDrugPrescribed = !prescribedCustomDrugs.isEmpty()
```

Reply:
Yep, changed to `val isAtLeastOneCustomDrugPrescribed = prescribedCustomDrugs.isNotEmpty()`